traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/ds/DeliveryService.java

```diff
@@ -125,6 +128,21 @@ public DeliveryService(final String id, final JsonNode dsJo) throws JsonUtilsExc
 		this.soa = dsJo.get("soa");
 		this.shouldAppendQueryString = JsonUtils.optBoolean(dsJo, "appendQueryString", true);
 
+		this.consistentHashQueryParams = new TreeSet<String>();
+		if (dsJo.has("consistentHashQueryParams")) {
+			final JsonNode cqpNode = dsJo.get("consistentHashQueryParams");
+			if (!cqpNode.isArray()) {
+				LOGGER.error("Delivery Service '" + id + "' has malformed consistentHashQueryParams. Disregarding.");
+			} else {
+				for (int i=0; i<cqpNode.size(); ++i) {
```

Comment:
Probably. I just didn't know they were iterable like that - the fact that they return `null` for indices greater than the array size rather than throwing an exception spooked me.